### ISSUE TITLE
Pull request query views using the Search API don't work in Codespaces

### ISSUE THREAD

**Issue by rolandcrosby-columntax** [microsoft/vscode-pull-request-github#5879]
- Extension version: 0.82.1
- VSCode Version: 1.87.2
  - Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
  - User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36
  - Embedder: codespaces
- Repository Clone Configuration (single repository/fork of an upstream repository): single repository
- Github Product (Github.com/Github Enterprise version x.x.x): Github.com

When using the Pull Requests extension in GitHub Codespaces, we are unable to access the pull request "queries" in the tree view ("Waiting For My Review", "Assigned To Me", "Created By Me"), regardless of what permissions we have granted to the GitHub auth token that is populated in the Codespace. We see the following error message when attempting to expand one of these "queries":

> `Fetching pull requests failed: Validation Failed: {"message":"The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.","resource":"Search","field":"q","code":"invalid"}`

For at least one user, this has led to an issue resembling what was described in #5874, where the user gets continuously prompted to sign in to GitHub; this seems to have eventually caused a separate error for that user involving GitHub API rate limiting.

Steps to Reproduce:

1. Set up a Codespace that has all possible permissions delegated to its GitHub repository via devcontainer.json (`"permissions": "write-all"` [here](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces#setting-additional-repository-permissions))
2. Open the pull requests extension, which is installed in Codespaces by default
3. Try to view any of the queries in the list of PRs like "Created By Me"

From the error message, it appears that this is caused by the extension's use of a "Search" API, which the token does not have permission to use. (The "All Open" and "Local Pull Request Branches" entries in the tree view work fine; I assume they don't enumerate PRs using the search API.)

Would it be possible to suppress the display of these "queries" in Codespaces, so users don't end up triggering GitHub API requests that will always fail?

**Comment by alexr00** [comment]
@rolandcrosby-columntax thanks for posting your work around. Since you didn't link a repository, I'm guessing this is a private, org owned, repo. I tried it out, and I can actually repro this issue, though I don't understand why yet. 

When in a Codespace, the auth token that we get comes directly from GitHub, instead of from our GitHub auth provider, and this isn't the first time we've seen something wrong with that token (https://github.com/microsoft/vscode-pull-request-github/issues/5325). I'll forward this issue to the GitHub folks. 

In the mean time, I'll see how we can avoid the loop that would case a rate limit to get hit.

**Comment by alexr00** [comment]
With https://github.com/microsoft/vscode-pull-request-github/pull/5887, the bug where the modal dialog shows repeatedly will be fixed. It should have just been an error notification.

**Comment by alexr00** [comment]
To verify:
1. Create a codespace from a private, org owned, repo.
2. Open the Pull Requests view
3. Expand one of the queries (other than "All")
4. Verify you don't end up in a re-auth loop.

**Comment by aeschli** [comment]
I still see this with 
Version: 1.89.0-insider
Commit: cc3429872df5e655abd12c233a9defbd15b0987d
Date: 2024-04-25T05:51:51.369Z
Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36 Edg/124.0.0.0

- create a codespace from vscode-copilot
- on open I get notifications:
```
Fetching pull requests failed: Validation Failed: {"message":"The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.","resource":"Search","field":"q","code":"invalid"}
```
- when expanding a pull request query I get a dialog

![image](https://github.com/microsoft/vscode-pull-request-github/assets/6461412/335479f9-ce28-4e39-9b8e-abe36584ab7a)

- following the dialog never ends

**Comment by alexr00** [comment]
@aeschli I'm not able to repro (I just get the expected error notification). You might not have had the latest GHPR pre-release version. Can you connect to your Codespace and make sure you don't have the "Reload" button next to GHPR in the extensions view?

**Comment by aeschli** [comment]
Yes, that is likely the problem. I'm codespaces web. It looks like I can't install the pre-release there (no way to select a version). Is there a trick?

**Comment by aeschli** [comment]
Hm, same problem from the desktop, the version seems to be fixed (v0.86.1)
